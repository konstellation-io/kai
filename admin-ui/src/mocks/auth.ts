import { GET_USER_EMAIL } from '../components/Header/Header.graphql';

export const usernameMock = {
  request: {
    query: GET_USER_EMAIL
  },
  result: {
    data: {
      me: {
        email: 'user@konstellation.com'
      }
    }
  }
};

export const unauthorizedUsernameMock = {
  request: {
    query: GET_USER_EMAIL
  },
  result: {
    data: {
      me: {
        email: 'unauthorizedUser@konstellation.com'
      }
    }
  },
  error: new Error('Unauthorized user')
};
