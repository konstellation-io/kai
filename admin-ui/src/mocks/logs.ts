import { GET_LOGS } from '../graphql/client/queries/getLogs.graphql';

export const currentLogPanelMock = {
  request: {
    query: GET_LOGS
  },
  result: {
    data: {
      runtimeId: 'runtimeIdMock',
      nodeId: 'nodeIdMock',
      nodeName: 'nodeNameMock'
    }
  }
};
