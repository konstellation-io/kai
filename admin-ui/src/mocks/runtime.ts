import { GET_DASHBOARD } from '../pages/Dashboard/Dashboard.graphql';
import { ADD_RUNTIME } from '../pages/AddRuntime/AddRuntime.graphql';

export const dashboardMock = {
  request: {
    query: GET_DASHBOARD
  },
  result: {
    data: {
      dashboard: {
        runtimes: [
          {
            id: '00001',
            status: 'ACTIVE',
            name: 'Some Name',
            creationDate: '2018-01-02'
          },
          {
            id: '00002',
            status: 'ACTIVE',
            name: 'Some Other Name',
            creationDate: '2018-02-03'
          }
        ],
        alerts: [
          {
            id: 'id0',
            type: 'error',
            message: 'some message',
            runtime: { id: '00001' }
          },
          {
            id: 'id1',
            type: 'error',
            message: 'some message 2',
            runtime: { id: '00001' }
          }
        ]
      }
    }
  }
};

export const addRuntimeMock = {
  request: {
    query: ADD_RUNTIME,
    variables: { name: 'New Runtime' }
  },
  result: {
    data: {
      createRuntime: {
        success: true,
        message: 'some message'
      }
    }
  }
};
