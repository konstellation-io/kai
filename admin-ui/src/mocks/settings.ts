import {
  GET_DOMAINS,
  GET_EXPIRATION_TIME,
  ADD_ALLOWED_DOMAIN,
  UPDATE_EXPIRATION_TIME
} from '../pages/Settings/Settings.graphql';
import { GET_USERS_ACTIVITY } from '../components/UserActivityList/UserActivityList.graphql';

export const domainMock = {
  request: {
    query: GET_DOMAINS
  },
  result: {
    data: {
      settings: {
        authAllowedDomains: ['domain.1', 'domain.2', 'domain.3.sample']
      }
    }
  }
};

export const usersActivityMock = {
  request: {
    query: GET_USERS_ACTIVITY
  },
  result: {
    data: {
      usersActivity: [
        {
          user: 'user1@domain.com',
          message: 'some message 1',
          date: '2019-01-01'
        },
        {
          user: 'user2@domain.com',
          message: 'some message 2',
          date: '2019-01-02'
        },
        {
          user: 'user3@domain.com',
          message: 'some message 3',
          date: '2019-01-03'
        }
      ]
    }
  }
};

export const expirationTimeMock = {
  request: {
    query: GET_EXPIRATION_TIME
  },
  result: {
    data: {
      settings: {
        cookieExpirationTime: 45
      }
    }
  }
};

export const addAllowedDomainMock = {
  request: {
    query: ADD_ALLOWED_DOMAIN,
    variables: { domainName: 'intelygenz.com' }
  },
  result: {
    data: {
      addAllowedDomain: {
        authAllowedDomains: [
          'domain.1',
          'domain.2',
          'domain.3.sample',
          'intelygenz.com'
        ]
      }
    }
  }
};

export const removeAllowedDomainMock = {
  request: {
    query: ADD_ALLOWED_DOMAIN,
    variables: { domainName: 'domain.3.sample' }
  },
  result: {
    data: {
      removeAllowedDomain: {
        authAllowedDomains: ['domain.1', 'domain.2']
      }
    }
  }
};

export const updateExpirationTime = {
  request: {
    query: UPDATE_EXPIRATION_TIME,
    variables: { input: { cookieExpirationTime: 10 } }
  },
  result: {
    data: {
      setSettings: {
        cookieExpirationTime: 12
      }
    }
  }
};
