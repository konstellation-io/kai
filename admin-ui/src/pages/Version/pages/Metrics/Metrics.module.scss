@import '../../../../styles/colors';

.container {
  position: relative;
  display: flex;
  flex-direction: column;
  height: 100%;
  width: 100%;
  min-width: 700px;
  overflow: auto;
  padding: 23px 0 10px 36px;
}
