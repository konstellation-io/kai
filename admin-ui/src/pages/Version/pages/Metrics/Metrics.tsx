import React, { useEffect, useState, useCallback } from 'react';
import DashboardTitle from './components/DashboardTitle/DashboardTitle';
import Charts from './components/Charts/Charts';
import Message from '../../../../components/Message/Message';
import { loader } from 'graphql.macro';
import styles from './Metrics.module.scss';
import {
  GetVersionConfStatus_versions,
  GetVersionConfStatus_runtime
} from '../../../../graphql/queries/types/GetVersionConfStatus';
import { useQuery } from '@apollo/react-hooks';
import {
  GetMetrics,
  GetMetricsVariables
} from '../../../../graphql/queries/types/GetMetrics';
import SpinnerCircular from '../../../../components/LoadingComponents/SpinnerCircular/SpinnerCircular';
import ErrorMessage from '../../../../components/ErrorMessage/ErrorMessage';
import { useParams } from 'react-router-dom';
import { VersionRouteParams } from '../../../../constants/routes';
import { useForm } from 'react-hook-form';
import moment, { Moment } from 'moment';

const GetMetricsQuery = loader(
  '../../../../graphql/queries/getMetrics.graphql'
);

type FormData = {
  startDate: Moment;
  endDate: Moment;
};

const DEFAULT_DATES: FormData = {
  startDate: moment()
    .subtract(30, 'days')
    .startOf('day'),
  endDate: moment().endOf('day')
};

type Props = {
  runtime?: GetVersionConfStatus_runtime;
  version?: GetVersionConfStatus_versions;
};

function Metrics({ runtime, version }: Props) {
  const { versionId, runtimeId } = useParams<VersionRouteParams>();
  const { data, loading, error, refetch } = useQuery<
    GetMetrics,
    GetMetricsVariables
  >(GetMetricsQuery, {
    variables: {
      versionId,
      runtimeId,
      startDate: DEFAULT_DATES.startDate.toISOString(),
      endDate: DEFAULT_DATES.endDate.toISOString()
    }
  });

  const { register, watch, setValue, handleSubmit } = useForm<FormData>({
    defaultValues: {
      startDate: DEFAULT_DATES.startDate,
      endDate: DEFAULT_DATES.endDate
    }
  });

  const [expanded, setExpanded] = useState<string>('');
  function toggleExpanded(nodeId: string): void {
    setExpanded(expanded ? '' : nodeId);
  }

  useEffect(() => {
    register({ name: 'startDate' });
    register({ name: 'endDate' });
  }, [register]);

  const summit = useCallback(() => {
    handleSubmit(({ startDate, endDate }: FormData) => {
      refetch({
        runtimeId,
        versionId,
        startDate: startDate.toISOString(),
        endDate: endDate.toISOString()
      });
    })();
  }, [handleSubmit, refetch, runtimeId, versionId]);

  // Submits the form every time 'endDate' is updated
  const endDate = watch('endDate');
  useEffect(() => {
    if (endDate) summit();
  }, [endDate, summit]);

  function getContent() {
    if (loading) return <SpinnerCircular />;
    if (error || !data) return <ErrorMessage />;
    if (data && data.metrics === null)
      return <Message text="There is no data for the selected dates" />;

    return (
      <Charts data={data} expanded={expanded} toggleExpanded={toggleExpanded} />
    );
  }

  return (
    <div className={styles.container}>
      <DashboardTitle
        runtimeName={runtime && runtime.name}
        versionName={version && version.name}
        value={watch}
        onChange={setValue}
        summit={summit}
      />
      {getContent()}
    </div>
  );
}

export default Metrics;
