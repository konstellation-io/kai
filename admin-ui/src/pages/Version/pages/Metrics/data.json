{
  "general": {
    "accuracy": {
      "total": 70,
      "micro": 30,
      "macro": 45,
      "weighted": 45
    },
    "null": 30,
    "labels": 45,
    "stdev": 45
  },
  "confusionMatrix": [
    { "x": "Repair Completed", "y": "Repair Completed", "value": 95 },
    { "x": "Repair Completed", "y": "Request Completed", "value": 56 },
    { "x": "Repair Completed", "y": "Disconnect Service", "value": 37 },
    { "x": "Repair Completed", "y": "Confirmed Issue Resolved", "value": 37 },
    { "x": "Repair Completed", "y": "Others", "value": 10 },
    {
      "x": "Repair Completed",
      "y": "Plan of Feature Change Completed",
      "value": 10
    },
    { "x": "Repair Completed", "y": "Resolved By Customer", "value": 10 },

    { "x": "Request Completed", "y": "Repair Completed", "value": 56 },
    { "x": "Request Completed", "y": "Request Completed", "value": 91 },
    { "x": "Request Completed", "y": "Disconnect Service", "value": 37 },
    { "x": "Request Completed", "y": "Confirmed Issue Resolved", "value": 37 },
    { "x": "Request Completed", "y": "Others", "value": 10 },
    {
      "x": "Request Completed",
      "y": "Plan of Feature Change Completed",
      "value": 10
    },
    { "x": "Request Completed", "y": "Resolved By Customer", "value": 10 },

    { "x": "Disconnect Service", "y": "Repair Completed", "value": 37 },
    { "x": "Disconnect Service", "y": "Request Completed", "value": 56 },
    { "x": "Disconnect Service", "y": "Disconnect Service", "value": 88 },
    { "x": "Disconnect Service", "y": "Confirmed Issue Resolved", "value": 37 },
    { "x": "Disconnect Service", "y": "Others", "value": 37 },
    {
      "x": "Disconnect Service",
      "y": "Plan of Feature Change Completed",
      "value": 10
    },
    { "x": "Disconnect Service", "y": "Resolved By Customer", "value": 10 },

    { "x": "Confirmed Issue Resolved", "y": "Repair Completed", "value": 37 },
    { "x": "Confirmed Issue Resolved", "y": "Request Completed", "value": 37 },
    { "x": "Confirmed Issue Resolved", "y": "Disconnect Service", "value": 42 },
    {
      "x": "Confirmed Issue Resolved",
      "y": "Confirmed Issue Resolved",
      "value": 78
    },
    { "x": "Confirmed Issue Resolved", "y": "Others", "value": 37 },
    {
      "x": "Confirmed Issue Resolved",
      "y": "Plan of Feature Change Completed",
      "value": 10
    },
    {
      "x": "Confirmed Issue Resolved",
      "y": "Resolved By Customer",
      "value": 10
    },

    { "x": "Others", "y": "Repair Completed", "value": 10 },
    { "x": "Others", "y": "Request Completed", "value": 37 },
    { "x": "Others", "y": "Disconnect Service", "value": 37 },
    { "x": "Others", "y": "Confirmed Issue Resolved", "value": 42 },
    { "x": "Others", "y": "Others", "value": 78 },
    { "x": "Others", "y": "Plan of Feature Change Completed", "value": 42 },
    { "x": "Others", "y": "Resolved By Customer", "value": 42 },

    {
      "x": "Plan of Feature Change Completed",
      "y": "Repair Completed",
      "value": 10
    },
    {
      "x": "Plan of Feature Change Completed",
      "y": "Request Completed",
      "value": 10
    },
    {
      "x": "Plan of Feature Change Completed",
      "y": "Disconnect Service",
      "value": 37
    },
    {
      "x": "Plan of Feature Change Completed",
      "y": "Confirmed Issue Resolved",
      "value": 37
    },
    { "x": "Plan of Feature Change Completed", "y": "Others", "value": 42 },
    {
      "x": "Plan of Feature Change Completed",
      "y": "Plan of Feature Change Completed",
      "value": 76
    },
    {
      "x": "Plan of Feature Change Completed",
      "y": "Resolved By Customer",
      "value": 76
    },

    { "x": "Resolved By Customer", "y": "Repair Completed", "value": 10 },
    { "x": "Resolved By Customer", "y": "Request Completed", "value": 10 },
    { "x": "Resolved By Customer", "y": "Disconnect Service", "value": 10 },
    {
      "x": "Resolved By Customer",
      "y": "Confirmed Issue Resolved",
      "value": 37
    },
    { "x": "Resolved By Customer", "y": "Others", "value": 42 },
    {
      "x": "Resolved By Customer",
      "y": "Plan of Feature Change Completed",
      "value": 42
    },
    { "x": "Resolved By Customer", "y": "Resolved By Customer", "value": 74 }
  ],
  "series": [
    {
      "title": "Accuracy",
      "data": [
        { "y": "Repair Completed", "x": 40 },
        { "y": "Request Completed", "x": 45 },
        { "y": "Disconnect Service", "x": 52 },
        { "y": "Confirmed Issue Resolved", "x": 32 },
        { "y": "Others", "x": 60 },
        { "y": "Plan of Feature Change Completed", "x": 72 },
        { "y": "Resolved By Customer", "x": 50 }
      ]
    },
    {
      "title": "Recall",
      "data": [
        { "y": "Repair Completed", "x": 80 },
        { "y": "Request Completed", "x": 25 },
        { "y": "Disconnect Service", "x": 12 },
        { "y": "Confirmed Issue Resolved", "x": 62 },
        { "y": "Others", "x": 70 },
        { "y": "Plan of Feature Change Completed", "x": 72 },
        { "y": "Resolved By Customer", "x": 30 }
      ]
    },
    {
      "title": "Support",
      "data": [
        { "y": "Repair Completed", "x": 20 },
        { "y": "Request Completed", "x": 25 },
        { "y": "Disconnect Service", "x": 12 },
        { "y": "Confirmed Issue Resolved", "x": 22 },
        { "y": "Others", "x": 50 },
        { "y": "Plan of Feature Change Completed", "x": 50 },
        { "y": "Resolved By Customer", "x": 10 }
      ]
    }
  ],
  "successVsFails": [
    { "x": "0:00", "y": 40 },
    { "x": "2:00", "y": 18 },
    { "x": "4:00", "y": 5 },
    { "x": "6:00", "y": 4 },
    { "x": "8:00", "y": 12 },
    { "x": "10:00", "y": 56 },
    { "x": "12:00", "y": 60 },
    { "x": "14:00", "y": 62 },
    { "x": "16:00", "y": 45 },
    { "x": "18:00", "y": 87 },
    { "x": "20:00", "y": 72 },
    { "x": "22:00", "y": 60 },
    { "x": "24:00", "y": 55 }
  ]
}
