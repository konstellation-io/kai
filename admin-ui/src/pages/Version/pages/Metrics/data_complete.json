{
  "general": {
    "accuracy": {
      "total": 70,
      "micro": 70,
      "macro": 67,
      "weighted": 69
    },
    "null": 5,
    "labels": 4,
    "stdev": 4
  },
  "confusionMatrix": [
    { "x": "low", "y": "low", "value": 84 },
    { "x": "low", "y": "mid", "value": 19 },
    { "x": "low", "y": "high", "value": 4 },
    { "x": "low", "y": "lux", "value": 3 },

    { "x": "mid", "y": "low", "value": 15 },
    { "x": "mid", "y": "mid", "value": 65 },
    { "x": "mid", "y": "high", "value": 36 },
    { "x": "mid", "y": "lux", "value": 12 },

    { "x": "high", "y": "low", "value": 1 },
    { "x": "high", "y": "mid", "value": 15 },
    { "x": "high", "y": "high", "value": 57 },
    { "x": "high", "y": "lux", "value": 47 },

    { "x": "lux", "y": "low", "value": 0 },
    { "x": "lux", "y": "mid", "value": 0 },
    { "x": "lux", "y": "high", "value": 3 },
    { "x": "lux", "y": "lux", "value": 38 }
  ],
  "series": [
    {
      "title": "Accuracy",
      "data": [
        { "y": "low", "x": 83 },
        { "y": "mid", "x": 62 },
        { "y": "high", "x": 59 },
        { "y": "lux", "x": 63 }
      ]
    },
    {
      "title": "Recall",
      "data": [
        { "y": "low", "x": 86 },
        { "y": "mid", "x": 66 },
        { "y": "high", "x": 58 },
        { "y": "lux", "x": 38 }
      ]
    },
    {
      "title": "Support",
      "data": [
        { "y": "low", "x": 3230 },
        { "y": "mid", "x": 2684 },
        { "y": "high", "x": 1501 },
        { "y": "lux", "x": 253 }
      ]
    }
  ],
  "successVsFails": [
    { "x": "0:00", "y": 40 },
    { "x": "2:00", "y": 18 },
    { "x": "4:00", "y": 5 },
    { "x": "6:00", "y": 4 },
    { "x": "8:00", "y": 12 },
    { "x": "10:00", "y": 56 },
    { "x": "12:00", "y": 60 },
    { "x": "14:00", "y": 62 },
    { "x": "16:00", "y": 45 },
    { "x": "18:00", "y": 87 },
    { "x": "20:00", "y": 72 },
    { "x": "22:00", "y": 60 },
    { "x": "24:00", "y": 55 }
  ]
}
