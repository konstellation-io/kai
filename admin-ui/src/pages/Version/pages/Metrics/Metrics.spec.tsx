import React from 'react';
import { shallow } from 'enzyme';
import Metrics from './Metrics';
import { version, runtime } from '../../../../mocks/version';

describe('Metrics', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Metrics runtime={runtime} version={version} />);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
