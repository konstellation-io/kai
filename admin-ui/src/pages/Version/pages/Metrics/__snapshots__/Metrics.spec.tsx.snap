// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Metrics matches snapshot 1`] = `
<div
  className="container"
>
  <DashboardTitle
    formSetValue={[Function]}
    formWatch={[Function]}
    runtimeName="runtimeName"
    versionName="versionName"
  />
  <Charts
    data={Object {}}
    expanded=""
    toggleExpanded={[Function]}
  />
</div>
`;
